test/e2e/quota/multi_tree.go

```diff
@@ -0,0 +1,301 @@
+			rootQuota1, err := f.SchedClientSet.SchedulingV1alpha1().ElasticQuotas(f.Namespace.Name).Get(context.TODO(), profile1.Spec.QuotaName, metav1.GetOptions{})
+			framework.ExpectNoError(err)
+
+			ginkgo.By("compare node 1 resource and root quota 1 min")
+			totalResource1 := corev1.ResourceList{}
+			totalResource1[corev1.ResourceCPU] = nodeList.Items[0].Status.Allocatable[corev1.ResourceCPU]
+			totalResource1[corev1.ResourceMemory] = nodeList.Items[0].Status.Allocatable[corev1.ResourceMemory]
+
+			gomega.Expect(quotav1.Equals(totalResource1, rootQuota1.Spec.Min)).Should(gomega.Equal(true))
+			framework.Logf("root quota min is right")
+			framework.Logf("node %v resource: %v, root quota 1 min: %v", nodeList.Items[0].Name, MarshalValue(nodeList.Items[0].Status.Allocatable), MarshalValue(rootQuota1.Spec.Min))
+
+			gomega.Expect(rootQuota1.Labels[extension.LabelQuotaTreeID]).ShouldNot(gomega.Equal(""))
+			gomega.Expect(rootQuota1.Labels[extension.LabelQuotaIsRoot]).Should(gomega.Equal("true"))
+			framework.Logf("root quota has tree id: %v, is root: %v", rootQuota1.Labels[extension.LabelQuotaTreeID], rootQuota1.Labels[extension.LabelQuotaIsRoot])
+
+			ginkgo.By("create child quota for root quota 1")
+			childQuota1 := &schedv1alpha1.ElasticQuota{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: f.Namespace.Name,
+					Name:      "profile1-child1-quota",
+					Labels: map[string]string{
+						QuotaE2eLabel:              "true",
+						extension.LabelQuotaParent: rootQuota1.Name,
+					},
+				},
+				Spec: schedv1alpha1.ElasticQuotaSpec{
+					Min: corev1.ResourceList{
+						corev1.ResourceCPU:    util.MultiplyQuant(rootQuota1.Spec.Min[corev1.ResourceCPU], 0.5),
+						corev1.ResourceMemory: util.MultiplyQuant(rootQuota1.Spec.Min[corev1.ResourceMemory], 0.5),
+					},
+					Max: rootQuota1.Spec.Min,
+				},
+			}
+
+			retChildQuota1, err := f.SchedClientSet.SchedulingV1alpha1().ElasticQuotas(f.Namespace.Name).Create(context.TODO(), childQuota1, metav1.CreateOptions{})
+			framework.ExpectNoError(err)
+
+			ginkgo.By("check child 1 quota tree id")
+			gomega.Expect(rootQuota1.Labels[extension.LabelQuotaTreeID]).Should(gomega.Equal(retChildQuota1.Labels[extension.LabelQuotaTreeID]), "parent and child has the same tree id")
+
+			ginkgo.By("check child 1 quota runtime quota")
+			gomega.Eventually(func() bool {
+				quota, err := f.SchedClientSet.SchedulingV1alpha1().ElasticQuotas(f.Namespace.Name).Get(context.TODO(), childQuota1.Name, metav1.GetOptions{})
+				framework.ExpectNoError(err)
+				runtime, err := extension.GetRuntime(quota)
+				framework.ExpectNoError(err)
+				if quotav1.Equals(runtime, quota.Spec.Min) {
+					framework.Logf("child 1 runtime quota is same to min, runtime: %v", MarshalValue(runtime))
+					return true
+				}
+				return false
+			}, 60*time.Second, 1*time.Second).Should(gomega.Equal(true))
+
+			ginkgo.By("create quota profile2 with resource ratio")
```

Comment:
There seem to be two scenarios, but they are both written together. It is recommended to split them into different cases.

Reply:
done